### PR TITLE
Issue #787

### PR DESCRIPTION
**Pull request by YoussefEgla**
- added 3 icons according to specifications
- aligned with code base convention

### CODE REVIEW

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import {HTMLAttributes} from 'react';
+import {MediumIconColor} from '../../../ignitus-Atoms/colors';
+
+type Props = HTMLAttributes<SVGElement>;
```
[the remark this answers is not in the document]
Comment (YoussefEgla) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx:5]: #789 Enable bracket spacing.
there is a conflict between prettier configuration and houndci.
to solve this issue bracketSpacing should be enabled in .prettierrc

**Comment by Dishebh** [comment]
@YoussefEgla please try mentioning the issue number you are fixing, and also kindly report in the issue description that you would like to take it. That way, someone else willing to take up the issue, can know it is already taken up.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      aria-label={title}
+      role="img"
+      viewBox="0 0 195 195"
+      xmlns="http://www.w3.org/2000/svg"
+      {...rest}
+    >
+      <title>{title}</title>
+      <rect
+        id="Rectangle-path"
+        fill={MediumIconColor}
```
Comment (Paarmita) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx:19]: Remove fill from all the icons.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts [hunk 3]
```diff
@@ -25,3 +25,6 @@ export const LinkedinIconColor: string = '#0e76a8';
 export const RedditIconColor: string = ' #FF4301';
 export const AngellistIconColor: string = '#040404';
 export const FacebookIconColor: string = ' #3b5998';
+export const SlackIconColor: string = '#000000';
```
[the remark this answers is not in the document]
Comment (YoussefEgla) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts:28]: It's better to reference color in same file according to structure to easily change it in the future

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/youtubeIcon/youtubeIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      aria-label={title}
+      role="img"
+      viewBox="0 0 176 124"
+      xmlns="http://www.w3.org/2000/svg"
+      {...rest}
+    >
+      <title>{title}</title>
+      <path
+        fill={YoutubeIconColor}
+        transform="translate(-8 -34)"
```
[the remark this answers is not in the document]
Comment (YoussefEgla) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/youtubeIcon/youtubeIcon.tsx:19]: - transform attribute is part of the svg exported from illustrator file provided an dcan't be removed.
- fill color from the file provided & easy reference

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        width="195"
```
[the remark this answers is not in the document]
Comment (YoussefEgla) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx:20]: - height and width attributes defined on the rectangle element inside the svg
- icon stretches to height and width of its parent unless else is specified

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Paarmita) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts:28]: But slack has many colors in the original icon and we will be either using black or white from the already defined colors so no need to define slack color separately.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx (hunk 2 above)
Comment (YoussefEgla) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx:19]: - referencing colors give developers the ability to change color easily in the future and has no harm

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (divyanshu-rawat) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx]: @YoussefEgla please have a look here this is how we assign properties to svg.

the {...rest} properties take care of assigning css to the `SVG` icon for example have a look here. 

https://github.com/Ignitus/Ignitus-client/blob/develop/src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-Navigation/styles.ts#L37

All these properties are passed as props to svg.

```
  width: 2.5rem;
  height: 2.5rem;
  path {
    fill: ${IgnitusBlue};
  }
```

So In summary, please don't assign any fill/height/width to `SVG` or any other css property.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx (hunk 2 above)
Comment (divyanshu-rawat) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/mediumIcon/mediumIcon.tsx]: Fill is not required at all.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (YoussefEgla) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts]: the icon provided has no predefined colors. please provide me with the one needed to make the change

**Comment by divyanshu-rawat** [comment]
@YoussefEgla you also need to icon [here](https://github.com/Ignitus/Ignitus-client/blob/develop/src/ignitus-Shared/ignitus-Utilities/Components/icon.tsx) & [here](https://github.com/Ignitus/Ignitus-client/blob/develop/src/ignitus-Shared/types/iconsTypes/iconEnums.ts) as well

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/youtubeIcon/youtubeIcon.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Paarmita) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/youtubeIcon/youtubeIcon.tsx:19]: It should be passed as a prop, please have a look at other icons of how we are doing that.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts [hunk 6]
```diff
@@ -25,3 +25,6 @@ export const LinkedinIconColor: string = '#0e76a8';
 export const RedditIconColor: string = ' #FF4301';
 export const AngellistIconColor: string = '#040404';
 export const FacebookIconColor: string = ' #3b5998';
+export const SlackIconColor: string = Black;
```
Comment (Paarmita) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts:28]: We don't need slack icon color, instead use Black mentioned as a fill property.

**Pull request closed without merging**